src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -1484,12 +1484,12 @@ private static string ParseName(string id, ref int index)
                 int delimiterOffset = id.IndexOfAny(s_nameDelimiters, index);
                 if (delimiterOffset >= 0)
                 {
-                    name = id.Substring(index, delimiterOffset - index);
+                    name = id[index..delimiterOffset];
                     index = delimiterOffset;
                 }
                 else
                 {
-                    name = id.Substring(index);
+                    name = id[index..];
```

Comment:
>name = id[index..]; [](http://example.com/codeflow?start=20&length=19)

This is a style only change. Please revert. BTW, it looks like the change results in worse IL.